validation/validator_test.go

```diff
@@ -185,6 +239,132 @@ func TestFails(t *testing.T) {
 	}
 }
 
+func TestCast(t *testing.T) {
+	type Data struct {
+		A string            `form:"a" json:"a"`
+		B int               `form:"b" json:"b"`
+		C int8              `form:"c" json:"c"`
+		D int16             `form:"d" json:"d"`
+		E int32             `form:"e" json:"e"`
+		F int64             `form:"f" json:"f"`
+		G uint              `form:"g" json:"g"`
+		H uint8             `form:"h" json:"h"`
+		I uint16            `form:"i" json:"i"`
+		J uint32            `form:"j" json:"j"`
+		K uint64            `form:"k" json:"k"`
+		L bool              `form:"l" json:"l"`
+		M float32           `form:"m" json:"m"`
+		N float64           `form:"n" json:"n"`
+		O []string          `form:"o" json:"o"`
+		P map[string]string `form:"p" json:"p"`
+	}
+
+	tests := []struct {
+		name       string
+		data       validate.DataFace
+		rules      map[string]string
+		expectData Data
+		expectErr  error
+	}{
+		{
+			name: "success when cast data",
+			data: func() validate.DataFace {
+				body := &Data{
+					A: "1",
+					B: 1,
+					C: 1,
+					D: 1,
+					E: 1,
+					F: 1,
+					G: 1,
+					H: 1,
+					I: 1,
+					J: 1,
+					K: 1,
+					L: true,
+					M: 1,
+					N: 1,
+					O: []string{"1"},
+					P: map[string]string{"a": "aa"},
+				}
+				jsonStr, err := json.Marshal(body)
+				assert.Nil(t, err)
+				request, err := http.NewRequest(http.MethodPost, "/", bytes.NewBuffer(jsonStr))
+				request.Header.Set("Content-Type", "application/json")
+				assert.Nil(t, err)
+				data, err := validate.FromRequest(request)
+				assert.Nil(t, err)
+
+				return data
+			}(),
+			rules: map[string]string{
+				"a": "required",
+				"b": "required",
+				"c": "required",
+				"d": "required",
+				"e": "required",
+				"f": "required",
+				"g": "required",
+				"h": "required",
+				"i": "required",
+				"j": "required",
+				"k": "required",
+				"l": "required",
+				"m": "required",
+				"n": "required",
+				"o": "required",
+				"p": "required",
+			},
+			expectData: Data{
+				A: "1",
```

Comment:
This feature seems to support `string => int`, right? Could you please add some such cases?

Reply:
> This feature seems to support `string => int`, right? Could you please add some such cases?

This already test in validation/validator_test.go L41-56

Reply:
Yes, we can add a test case for `TestBind`, but this case belongs to the `CastValue` method directly, so we'd better add the case to `TestCastValue`, too.